### PR TITLE
code_format: fix path matching

### PR DESCRIPTION
**Pull request by DavidPeet8**
**Commit Message:** This change fixes an issue in the code formatter where explicitly
excluded paths are incorrectly matched when the script is provided
arguments that do not start with './' or that end with a trailing
slash.

To address this issue this change normalizes paths before checking
if the path is excluded.

**Additional Description:** I think the core issue here is the use of direct string matching for excluded files. Refactoring to address the root cause is out of scope for this change, but will likely need to be done in the future.

**Risk Level:** low

**Testing:** Ran code format with arguments of the form 'path/to/dir' and './path/to/dir/', verifying that excluded files are not formatted as expected.

**Docs Changes:** n/a
**Release Notes:** n/a
**Platform Specific Features:** n/a
**Fixes:** #20077

### CODE REVIEW

**Comment by daixiang0** [comment]
Thanks for your contribution!

Please wait for a maintainer to trigger CI flow.

/cc @envoyproxy/stable-maintainers

File: tools/code_format/check_format.py
Comment (yanavlasov) [inline tools/code_format/check_format.py:1088]: Would it make sense to prepend "./" before checking if it is a directory? Just in case.

**Review by yanavlasov — COMMENTED** [review]
/wait-any

File: tools/code_format/check_format.py
Comment (phlax) [reply tools/code_format/check_format.py:1088]: im wondering if `pathlib` could be helpful here - it does quite a bit of path normalization - altho it seems like elsewhere is interpreting the appended `/` so maybe that just complicates things

Comment (DavidPeet8) [reply tools/code_format/check_format.py:1088]: > Would it make sense to prepend "./" before checking if it is a directory? Just in case.

@yanavlasov - Good point, calling this with an absolute path would lead to unexpected behaviour

> im wondering if `pathlib` could be helpful here - it does quite a bit of path normalization - altho it seems like elsewhere is interpreting the appended `/` so maybe that just complicates things

@phlax - I think in the future something like `pathlib` could be quite helpful, but I think the introduction of path normalization via `pathlib` or some other library function should come with a larger refactor in this file. In particular, I think a diff changing to normalize via a library would be a good opportunity to refactor how paths are handled in general across the formatter. 

As you mentioned, the trailing slash becomes a bit awkward to deal with as we would be modifying the path after it is normalized - hopefully this could be solved with such an additional refactor

**Comment by DavidPeet8** [comment]
@phlax - Are you able to trigger the pending workflow?

**Review by phlax — APPROVED** [review]
lgtm, thanks @DavidPeet8  for working through this

**Pull request merged**
